src/main/java/com/spotify/futures/CompletableFuturesExtra.java

```diff
@@ -164,7 +164,9 @@ private CompletableFuturesExtra() {
     try {
       return future.join();
     } catch (CompletionException e) {
-      throw Throwables.propagate(e.getCause());
+      final Throwable throwable = e.getCause();
+      Throwables.throwIfUnchecked(throwable);
+      throw new RuntimeException(throwable);
```

Comment:
maybe re-throwing original `CompletionException` will be better here?

Reply:
Yes, will change.